sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -106,7 +106,7 @@ abstract class Attribute extends LeafExpression with NamedExpression with NullIn
 
   override def toAttribute: Attribute = this
   def newInstance(): Attribute
-
+  def stringWithType: String = s"$toString: ${dataType.simpleString}"
```

Comment:
Nit: the `:` isn't needed and will accumulate to quite a few extra characters in plans. SQL uses `(id bigint, data string)` without `:` so I think we should mirror that.